python_utils.py

```diff
@@ -374,6 +377,54 @@ def convert_to_bytes(string_to_convert):
     Returns:
         bytes. The encoded string.
     """
-    if isinstance(string_to_convert, STR):
+    if isinstance(string_to_convert, UNICODE):
         return string_to_convert.encode('utf-8')
     return bytes(string_to_convert)
+
+
+def _recursively_convert_to_str(value):
+    """Convert all builtins.bytes and builtins.str elements in a data structure
+    to bytes and unicode respectively. This is required for the
+    yaml.safe_dump() function to work as it only works for unicode and bytes and
+    not builtins.bytes nor builtins.str(UNICODE). See:
+    https://stackoverflow.com/a/1950399/11755830
+
+    Args:
+        value: list|dict|BASESTRING. The data structure to convert.
```

Comment:
Just call this a json-like.

Reply:
Actually, I prefer to leave this one as-is. I think I said earlier in another PR that "json-like" is fine, but on reflection "json" is actually a string encoding so this is ambiguous.